pkg/kubelet/kubelet.go

```diff
@@ -2473,6 +2473,17 @@ func (kl *Kubelet) syncLoopIteration(ctx context.Context, configCh <-chan kubety
 			status = "started"
 		}
 		handleProbeSync(kl, update, handler, "startup", status)
+	case update := <-kl.containerManager.Updates():
+		if pod, ok := kl.podManager.GetPodByUID(types.UID(update.PodUID)); ok {
+			klog.V(3).InfoS("SyncLoop (containermanager): event for pod", "pod", klog.KObj(pod), "event", update)
+			// Updating the pod by syncing it again
+			// We do not apply the optimization by updating the status directly, but can do it later
+			handler.HandlePodSyncs([]*v1.Pod{pod})
```

Comment:
so HandlePodSyncs works in batches, but we're updating one by one. It could make sense in a later PR to do some batching here

Reply:
added batching